### PR TITLE
test: add e2e tests for 1st step of create colony flow

### PR DESCRIPTION
**Pull request by olgapod**
## Description

I've broken down the initial PR into smaller parts to help identify the cause of the inconsistent test results. This is the first part covering the 1st step of the Create colony form. 

- Added e2e tests for the Create Colony flow / Details step
- Set up playwright eslint plugin

## Testing

**Prerequisites**: The test script simulates user interactions with the local instance of the web app, which means that the BE is expected to be set up and running. For now, backend services should be started manually and the initial test data should be seeded . Therefore, ensure that the development environment is properly set up by following the instructions in the README file

- Start your local dev environment  `npm run dev` 
- Seed the mock data `node scripts/create-data.js` 
- If it's the first time you run e2e tests on your machine - run `npm e2e:install` 

- run `npm run e2e` or `npm run e2e:ui`
- Ensure all tests pass

## Diffs

**New stuff** ✨

- playwright folder is added to the lint script
- added loading of the .env file into playwright environment

### CODE REVIEW

File: playwright/utils/common.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  timeout = 500,
+}: {
+  page: Page;
+  label: string | RegExp;
+  value: string;
+  timeout?: number;
+}) => {
+  await page.getByLabel(label).pressSequentially(value, { delay: 100 });
+  // eslint-disable-next-line playwright/no-wait-for-timeout
+  await page.waitForTimeout(timeout);
```
Comment (iamsamgibbs) [inline playwright/utils/common.ts:29]: Adding an extra timeout before pressSequentially seemed to fix things for me.

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
These are still failing for me unfortunately! But I found adding an extra timeout fixed it. I've left a comment. :) With the extra timeout, they are even passing when running in parallel.

File: playwright/utils/common.ts [hunk 2]
```diff
@@ -12,3 +12,19 @@ export const acceptCookieConsentBanner = async (page: Page) => {
...
+};
```
Comment (mmioana) [inline playwright/utils/common.ts:30]: Really nice you extracted this helper function ❤️

**Review by mmioana — CHANGES_REQUESTED** [review]
Really nice and hard work @olgapod 🙌 thank you so much for your continuous effort to make these tests run flawlessly 🫶 

I kept getting these 2 tests failing
Screenshot 2024-10-31 at 10 08 32

From my debugging it seemed the input fields were not correctly focusing, but these lines in `fillInputByLabelWithDelay` seemed to do the trick
```
  await page.getByLabel(label).focus();

  await page.waitForTimeout(timeout);

  await page.getByLabel(label).pressSequentially(value, { delay: 100 });
  // eslint-disable-next-line playwright/no-wait-for-timeout
  await page.waitForTimeout(timeout);
```

File: playwright/utils/common.ts (hunk 1 above)
Comment (olgapod) [reply playwright/utils/common.ts:29]: thank you @iamsamgibbs  for your help with that. I’ll add an extra timeout for now, although normally, it shouldn’t be necessary. My assumption is that this issue might be related to the app running in dev mode - e2e tests should be run against the optimized build, with no extra overhead of the dev server, HMR, huge js/css chunks, etc. Playwright, due to its ability to interact with the app very quickly, may be revealing a problem that’s hard to reproduce manually. 
Besides that, I’ve found there could indeed be an issue with at least one particular input field's validation - I was able to reproduce it manually. Here’s the link to the issuehttps://github.com/JoinColony/colonyCDapp/issues/3565

Comment (Nortsova) [reply playwright/utils/common.ts:32]: but do we actually need timeout after pressSequentially ?

File: playwright/utils/create-colony.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  await page.getByText(/dev wallet 2/i).click();
+
+  await page.getByText(/Local Ganache Instance/i).waitFor({ state: 'visible' });
+
+  const loadingIndicator = page.getByText(/Checking your access.../i);
+  const loadingIndicatorVisible = await loadingIndicator.isVisible();
+
+  if (loadingIndicatorVisible) {
+    await loadingIndicator.waitFor({ state: 'hidden' });
+  }
```
Comment (Nortsova) [inline playwright/utils/create-colony.ts:62]: good staff 🙌  I like this utils 😄

File: playwright/utils/graphqlHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
+import { request } from '@playwright/test';
+
+const API_KEY = 'da2-fakeApiId123456';
+const GRAPHQL_URI =
+  process.env.AWS_APPSYNC_GRAPHQL_URL || 'http://localhost:20002/graphql';
```
Comment (Nortsova) [inline playwright/utils/graphqlHelpers.ts:5]: let's revisit these keys, can we move it to `.env` or this one is expected?

File: playwright/e2e/create-colony.spec.ts [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        expect(
+          page
+            .getByTestId('onboarding-subheading')
+            .filter({ hasText: /Let’s set up your Colony/i }),
+        ).toBeVisible(),
+        expect(page.getByLabel(/colony name/i)).toBeVisible(),
+        expect(page.getByLabel(/custom colony URL/i)).toBeVisible(),
+        expect(page.getByRole('button', { name: /continue/i })).toBeVisible(),
+      ]);
+      // Sidebar navigation and its components are visible
```
Comment (Nortsova) [inline playwright/e2e/create-colony.spec.ts:43]: I like this refactored stuff 😍  Now tests work faster and in parallel 🙌

**Review by Nortsova — APPROVED** [review]
Hey @olgapod , great work on this 🙌  Especially considering that you establish tests from scratch and faced with an unexpected bugs, you are a superhero 🦸‍♀️ 

I tried 10 times and it passed:
image

The only concern that I have is API keys 🙌 

Great work on this 🔥

File: playwright/utils/create-colony.ts (hunk 3 above)
Comment (olgapod) [reply playwright/utils/create-colony.ts:62]: > but do we actually need timeout after pressSequentially ?

I'm afraid the only way to see it is to remove it and run on your and other devs machines. ( in my case neither of them is needed). Let me try to remove it.

Comment (olgapod) [reply playwright/utils/create-colony.ts:62]: most probably once we properly build the app and also fix the bug with quick typing and validation I mentioned above, there will be no need for any of the timeouts)

File: playwright/utils/graphqlHelpers.ts (hunk 4 above)
Comment (olgapod) [reply playwright/utils/graphqlHelpers.ts:5]: > let's revisit these keys, can we move it to .env or this one is expected?

it's copied from the scripts we already have  in, e.g. `scripts/create-colony.ts`

File: playwright/utils/common.ts (hunk 1 above)
Comment (olgapod) [reply playwright/utils/common.ts:32]: ok let me remove it )

Comment (olgapod) [reply playwright/utils/common.ts]: @Nortsova removed it, is all tests still green for you?

**Review by iamsamgibbs — APPROVED** [review]
Nice job with all the back and forth on this! All tests are passing consistently for me now. 🎉 

Screenshot 2024-11-01 at 09 30 02

Screenshot 2024-11-01 at 09 32 52

**Review by mmioana — APPROVED** [review]
Amazing effort on the tests setup and initial flow @olgapod ❤️ 

All tests run without an issue now ✨ 

Screenshot 2024-11-01 at 12 04 44

🎉

**Pull request merged**